src/python/zquantum/core/wip/estimators/estimation.py

```diff
@@ -0,0 +1,236 @@
+from typing import List, Tuple, Optional
+from openfermion import QubitOperator, IsingOperator
+import pyquil
+import numpy as np
+
+from ...circuit._circuit import Circuit
+from .estimation_interface import (
+    EstimationTask,
+    EstimationTaskTransformer,
+)
+from ...hamiltonian import group_comeasureable_terms_greedy, estimate_nmeas_for_frames
+from ...utils import scale_and_discretize
+from .estimation_interface import EstimationTask
+from ...interfaces.backend import QuantumBackend, QuantumSimulator
+from ...measurement import (
+    ExpectationValues,
+    expectation_values_to_real,
+    concatenate_expectation_values,
+)
+
+greedy_grouping_with_context_selection: EstimationTaskTransformer
+uniform_shot_allocation: EstimationTaskTransformer
+proportional_shot_allocation: EstimationTaskTransformer
```

Comment:
what's the purpose of these lines?

Reply:
FYI: the original point of Python's protocols was that you don't need to explicitly mark the code that conforms to a one, you can just use the protocol as a type hint on the use site and the type checker would still utilize that. However, explicit marks can still make sense for the sake of documentation and indeed this makes sense here IMHO.

Reply:
@mstechly We determined that it shouldn't be there, as it trips type checkers. Unfortunately, there seems to be no way for marking functions implementing protocols explicitly.

Reply:
That's what Code Review is for :D 
Fixed in https://github.com/zapatacomputing/z-quantum-core/pull/259/commits/a65bd21d57a88fcd08eebfe7e826aa99b86e6792